config/grafana/config.go

```diff
@@ -28,6 +28,101 @@ func ConfigureOrgIDRefs(p *ujconfig.Provider) {
 	}
 }
 
+// ConfigureOnCallRefsAndSelectors add reference and selector fields for Grafana OnCall resources
+func ConfigureOnCallRefsAndSelectors(p *ujconfig.Provider) {
+	escalationChainRef := ujconfig.Reference{
+		TerraformName:     "grafana_oncall_escalation_chain",
+		RefFieldName:      "EscalationChainRef",
+		SelectorFieldName: "EscalationChainSelector",
+	}
+	integrationRef := ujconfig.Reference{
+		TerraformName:     "grafana_oncall_integration",
+		RefFieldName:      "IntegrationRef",
+		SelectorFieldName: "IntegrationSelector",
+	}
+	oncallShiftRef := ujconfig.Reference{
+		TerraformName:     "grafana_oncall_on_call_shift",
+		RefFieldName:      "OnCallShiftRef",
+		SelectorFieldName: "OnCallShiftSelector",
+	}
+	outgoingWebhookRef := ujconfig.Reference{
+		TerraformName:     "grafana_oncall_outgoing_webhook",
+		RefFieldName:      "OutgoingWebhookRef",
+		SelectorFieldName: "OutgoingWebhookSelector",
+	}
+	scheduleRef := ujconfig.Reference{
+		TerraformName:     "grafana_oncall_schedule",
+		RefFieldName:      "ScheduleRef",
+		SelectorFieldName: "ScheduleSelector",
+	}
+
+	p.AddResourceConfigurator("grafana_oncall_escalation", func(r *ujconfig.Resource) {
+		r.References["escalation_chain_id"] = escalationChainRef
+		r.References["action_to_trigger"] = outgoingWebhookRef
+		r.References["notify_on_call_from_schedule"] = scheduleRef
+		// r.References["group_to_notify"] = oncallUserGroupRef
+		// r.References["notify_to_team_members"] = oncallTeamRef
+		// r.References["persons_to_notify"] = oncallUserRef
+		// r.References["persons_to_notify_next_each_time"] = oncallUserRef
+	})
+
+	p.AddResourceConfigurator("grafana_oncall_integration", func(r *ujconfig.Resource) {
+		r.References["default_route.escalation_chain_id"] = escalationChainRef
+		// r.References["team_id"] = oncallTeamRef
+	})
+
+	p.AddResourceConfigurator("grafana_oncall_route", func(r *ujconfig.Resource) {
+		r.References["escalation_chain_id"] = escalationChainRef
+		r.References["integration_id"] = integrationRef
+	})
+
+	p.AddResourceConfigurator("grafana_oncall_schedule", func(r *ujconfig.Resource) {
+		// TODO: this should be a list.. will this work like this?
+		// for reference https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/oncall_schedule#shifts
+		r.References["shifts"] = oncallShiftRef
```

Comment:
For fields with special names like this, I wouldn't call it `OnCallShiftRef`. It'd call it `ShiftsRef` or something, so that it's clear to which field it's mapped

As it is, the generated CRD looks like this:
```
...
onCallShiftRef:
                    description: References to OnCallShift in oncall to populate shifts.
                    items:
                      description: A Reference to a named object.
                      properties:
                        name:
                          description: Name of the referenced object.
                          type: string
                        policy:
                          description: Policies for referencing.
                          properties:
                            resolution:
                              default: Required
                              description: |-
                                Resolution specifies whether resolution of this reference is required.
                                The default is 'Required', which means the reconcile will fail if the
                                reference cannot be resolved. 'Optional' means this reference will be
                                a no-op if it cannot be resolved.
                              enum:
                              - Required
                              - Optional
                              type: string
                            resolve:
                              description: |-
                                Resolve specifies when this reference should be resolved. The default
                                is 'IfNotPresent', which will attempt to resolve the reference only when
                                the corresponding field is not present. Use 'Always' to resolve the
                                reference on every reconcile.
                              enum:
                              - Always
                              - IfNotPresent
                              type: string
                          type: object
                      required:
                      - name
                      type: object
                    type: array
                  onCallShiftSelector:
                    description: Selector for a list of OnCallShift in oncall to populate
                      shifts.
                    properties:
                      matchControllerRef:
                        description: |-
                          MatchControllerRef ensures an object with the same controller reference
                          as the selecting object is selected.
                        type: boolean
                      matchLabels:
                        additionalProperties:
                          type: string
                        description: MatchLabels ensures an object with matching labels
                          is selected.
                        type: object
                      policy:
                        description: Policies for selection.
                        properties:
                          resolution:
                            default: Required
                            description: |-
                              Resolution specifies whether resolution of this reference is required.
                              The default is 'Required', which means the reconcile will fail if the
                              reference cannot be resolved. 'Optional' means this reference will be
                              a no-op if it cannot be resolved.
                            enum:
                            - Required
                            - Optional
                            type: string
                          resolve:
                            description: |-
                              Resolve specifies when this reference should be resolved. The default
                              is 'IfNotPresent', which will attempt to resolve the reference only when
                              the corresponding field is not present. Use 'Always' to resolve the
                              reference on every reconcile.
                            enum:
                            - Always
                            - IfNotPresent
                            type: string
                        type: object
                    type: object
                  shifts:
                    description: |-
                      call shifts.
                      The list of ID's of on-call shifts.
                    items:
                      type: string
                    type: array
                    x-kubernetes-list-type: set
...
```